agent/containermetadata/utils.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	metadataJoinSuffix = "metadata"
+)
+
+// getTaskIDfromARN parses a task ARN and produces the task ID
+// A task ARN has format arn:aws:ecs:[region]:[account-id]:task/[task-id]
+// For a correctly formatted ARN we split it over ":" into 6 parts, the last part
+// containing the task-id which we extract by splitting it by "/".
+// This function should be eventually be replaced by a standardized ARN parsing
+// module
+func getTaskIDfromARN(taskARN string) (string, error) {
```

Comment:
We should be able to get rid of this if we use the updated SDK and the `arn.Parse()` method from https://github.com/aws/aws-sdk-go/commit/1fb10cfa0eea8817b40942b70889cb1509479527

Reply:
fixed. updating sdk here https://github.com/aws/amazon-ecs-agent/pull/1006 and added the arn to this set of changes.

Reply:
not including this change since godep is doing [this](https://github.com/tools/godep/issues/498#issuecomment-238946586).